### PR TITLE
Remove relabelling rules from service monitors when MCAC is disabled

### LINKED ISSUE #822: Remove relabelling rules from service monitors when MCAC is disabled
We have to remove the relabelling rules from the service monitors when MCAC is disabled, as the metrics will be correctly labeled in the agent.

Also, configure the service monitor to point to the new metrics endpoint URL: `http://127.0.0.1:9000/metrics`

### PR DESCRIPTION
**Pull request by adutra**
**What this PR does**:
* Remove relabelling rules from service monitors when MCAC is disabled
* Change ServiceMonitor target port when MCAC is disabled

**Which issue(s) this PR fixes**:
Fixes #822

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pkg/telemetry/prom_cass_servicemonitor.go [hunk 1]
```diff
@@ -13,17 +13,22 @@ import (
...
+// Static configuration for ServiceMonitor's endpoints when using the legacy MCAC stack (Metrics
+// Collector for Apache Cassandra).
+const endpointStringLegacy = `
 apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 spec:
   endpoints:
-  - port: prometheus
-    interval: 15s
+  - interval: 15s
```
Comment (adutra) [inline pkg/telemetry/prom_cass_servicemonitor.go:31]: There was something fishy here. We were defining both `port` and `targetPort` but the specs say we should define one or the other, not both. Besides, `prometheus` port is 9090, which is not what we want in any case (MCAC enabled or disabled). So I went ahead and removed `port: prometheus`. Unfortunately there is no e2e or kuttl test to verify this, so I'm going to do some manual testing around this today.

File: pkg/telemetry/prometheus_resourcer.go [hunk 2]
```diff
@@ -25,8 +25,14 @@ type PrometheusResourcer struct {
 }
 
 func (cfg PrometheusResourcer) validate() error {
-	if cfg.MonitoringTargetNS == "" || cfg.MonitoringTargetName == "" || cfg.ServiceMonitorName == "" {
-		return TelemetryConfigIncomplete{}
+	if cfg.MonitoringTargetNS == "" {
```
Comment (adutra) [inline pkg/telemetry/prometheus_resourcer.go:28]: Not directly related to the issue at hand, but this was returning incomplete error messages and making my unit tests fail.

File: pkg/telemetry/prometheus_resourcer.go [hunk 3]
```diff
@@ -39,7 +45,7 @@ func (cfg PrometheusResourcer) UpdateResources(
 	desiredSM *promapi.ServiceMonitor,
 ) error {
 	if err := cfg.validate(); err != nil {
-		return TelemetryConfigIncomplete{}
+		return err
```
Comment (adutra) [inline pkg/telemetry/prometheus_resourcer.go:48]: Unrelated change, but this seemed wrong.

File: pkg/telemetry/prometheus_resourcer_test.go [hunk 4]
```diff
@@ -34,11 +34,11 @@ func Test_PrometheusResourcer_UpdateResources_Create_CassDC(t *testing.T) {
 		CommonLabels:         map[string]string{k8ssandraapi.K8ssandraClusterNameLabel: "test-k8ssandracluster"},
 	}
 	ownerCassDC := test.NewCassandraDatacenter("test-cassdc", "test-namespace")
-	serviceMonitor, err := cfg.NewCassServiceMonitor()
+	serviceMonitor, err := cfg.NewCassServiceMonitor(true)
 	if err != nil {
 		assert.Fail(t, "couldn't create new ServiceMonitor for CassDC", "error", err)
 	}
-	if cfg.UpdateResources(ctx, fakeClient, &ownerCassDC, &serviceMonitor); err != nil {
+	if err := cfg.UpdateResources(ctx, fakeClient, &ownerCassDC, serviceMonitor); err != nil {
```
Comment (adutra) [inline pkg/telemetry/prometheus_resourcer_test.go:41]: Blatant error that the compiler happily compiles...

File: pkg/telemetry/prom_cass_servicemonitor.go [hunk 5]
```diff
@@ -27,7 +25,6 @@ spec:
     path: /metrics
     scheme: http
     scrapeTimeout: 15s
-    targetPort: %d
```
Comment (adutra) [inline pkg/telemetry/prom_cass_servicemonitor.go:30]: OK I think I understand this now. Looking at how `port` and `targetPort` [are handled](https://github.com/prometheus-operator/prometheus-operator/blob/v0.28.0/pkg/prometheus/promcfg.go#L378-L399),  we don't need `targetPort`. `port` most be a named port exposed by the all-pods service, so here `prometheus` is indeed a named port, and corresponds to 9103.

For the new, modern stack, we just need to switch to the new named port: `metrics` which corresponds to 9000.

File: pkg/telemetry/prometheus_resourcer_test.go (hunk 4 above)
Comment (adejanovski) [reply pkg/telemetry/prometheus_resourcer_test.go:41]: 😅 
The compiler is very kind to us.

**Review by adejanovski — APPROVED** [review]
Works as expected and fine looking code as always 👍

**Pull request merged**